### PR TITLE
tests: enhance xtimer_msg and add auto test script

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

This PR enhances the xtimer_msg tests, by a) removing timex and use xtimer only instead, and b) reduce memory size by lowering thread stack size to an appropriate
 (and still more than enough) value. 

Further, this PR adds a test runner script for automation.

### Issues/PRs references

followup on #7871, and others

### CODE REVIEW

File: tests/xtimer_msg/main.c [hunk 1]
```diff
@@ -38,13 +38,14 @@ struct timer_msg {
     msg_t msg;
 };
 
-struct timer_msg msg_a = { .interval = 2*(1000000), .text = "Hello World", };
-struct timer_msg msg_b = { .interval = 5*(1000000), .text = "This is a Test" };
+struct timer_msg msg_a = {  .interval = (2 * US_PER_SEC),
```
Comment (aabadie) [inline tests/xtimer_msg/main.c]: 2 spaces after `{`, I think 1 is enough ;)

File: tests/xtimer_msg/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import sys
+
+
+def testfunc(child):
+
```
Comment (aabadie) [inline tests/xtimer_msg/tests/01-run.py]: This blank line could be removed

File: tests/xtimer_msg/tests/01-run.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    # 1st check for periodic 2s Hello World message, i.e., 2 output + 1 msg
+    for i in range(7):
```
Comment (aabadie) [inline tests/xtimer_msg/tests/01-run.py]: since `i` is not used, you can replace it with `_`. Not blocking btw.

File: tests/xtimer_msg/main.c (hunk 1 above)
Comment (smlng) [reply tests/xtimer_msg/main.c]: yeah, I aligned `.interval` and `.text` below on tabs with size 4, but typically we only have 1 space here, will adapt.

**Comment by smlng** [comment]
addressed comments, thx for reviewing @aabadie

File: tests/xtimer_msg/main.c [hunk 4]
```diff
@@ -38,13 +38,14 @@ struct timer_msg {
     msg_t msg;
 };
 
-struct timer_msg msg_a = { .interval = 2*(1000000), .text = "Hello World", };
-struct timer_msg msg_b = { .interval = 5*(1000000), .text = "This is a Test" };
+struct timer_msg msg_a = { .interval = (2 * US_PER_SEC),
+                           .text = "Hello World", };
```
Comment (aabadie) [inline tests/xtimer_msg/main.c]: Very minor, there's a trailing comma before `}`. I could be removed to be consistent the other struct (`timer_msg msg_b`).

Comment (smlng) [reply tests/xtimer_msg/main.c]: oops missed that, will change in a second 😄

**Comment by smlng** [comment]
done

File: tests/xtimer_msg/tests/01-run.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import os
+import sys
+
+
+def testfunc(child):
+    # 1st check for periodic 2s Hello World message, i.e., 2 output + 1 msg
+    for _ in range(7):
+        child.expect(r"sec=\d+ min=\d+ hour=\d+")
+        child.expect(r"sec=\d+ min=\d+ hour=\d+")
+        child.expect(r"now=\d+:\d+ -> every 2.0s: Hello World")
```
Comment (aabadie) [inline tests/xtimer_msg/tests/01-run.py]: Just thinking of it: what about adding some high level verification of the timing. Using python `time` module I think it should be possible to check that the messages are effectively displayed every 2s (and 5s below).

Comment (kaspar030) [reply tests/xtimer_msg/tests/01-run.py]: Our (reviewing guidelines)[https://github.com/RIOT-OS/RIOT/wiki/Guidelines-for-Reviewing-a-Pull-Request) state clearly that "while you're at it" is bad practice. Get this merged, improve later!

Also, there was consensus that we check (x)timer's accuracy in *one* test designed for that. All other tests then trust the node's notion of time. Otherwise we (unneccessarily) complicate the tests.

Comment (smlng) [reply tests/xtimer_msg/tests/01-run.py]: @kaspar030, I don't think "while you're at it" applies here: the script is added here, so we why not make it "right" from the start. On the other hand I also agree with you, that we don't want to verify xtimer accuracy here but rather that messages arrive in a certain, expected pattern. Which should be sufficiently verified by looking for 2 + 1 "Hello world" message in the first case and 5 + 1 "This is a Test" messages in the second case.

TL;DR: lets leave as is for now.

Comment (aabadie) [reply tests/xtimer_msg/tests/01-run.py]: This was not a blocking comment from my side but rather a question. Thanks for the answers :)
